data/transactions/logic/doc.go

```diff
@@ -24,91 +24,97 @@ import (
+	"gaids":         "push the ID of the asset or application created in the Xth transaction of the current group",
+	"bnz":           "branch to TARGET if value X is not zero",
+	"bz":            "branch to TARGET if value X is zero",
+	"b":             "branch unconditionally to TARGET",
+	"return":        "use last value on stack as success value; end",
+	"pop":           "discard value X from stack",
+	"dup":           "duplicate last value on stack",
+	"dup2":          "duplicate two last values on stack: A, B -> A, B, A, B",
+	"dig":           "push the Nth value from the top of the stack. dig 0 is equivalent to dup",
+	"swap":          "swaps two last values on stack: A, B -> B, A",
+	"select":        "selects one of two values based on top-of-stack: A, B, C -> (if C != 0 then B else A)",
+	"concat":        "pop two byte-arrays A and B and join them, push the result",
+	"substring":     "pop a byte-array A. For immediate values in 0..255 S and E: extract a range of bytes from A starting at S up to but not including E, push the substring result. If E < S, or either is larger than the array length, the program fails",
+	"substring3":    "pop a byte-array A and two integers B and C. Extract a range of bytes from A starting at B up to but not including C, push the substring result. If C < B, or either is larger than the array length, the program fails",
+	"getbit":        "pop a target A (integer or byte-array), and index B. Push the Bth bit of A.",
+	"setbit":        "pop a target A, index B, and bit C. Set the Bth bit of A to C, and push the result",
+	"getbyte":       "pop a byte-array A and integer B. Extract the Bth byte of A and push it as an integer",
+	"setbyte":       "pop a byte-array A, integer B, and small integer C (between 0..255). Set the Bth byte of A to C, and push the result",
+	"extract":       "pop a byte-array A. For immediate values in 0..255 S and L: extract a range of bytes from A starting at S up to but not including S+L, push the substring result. If S or S+L is larger than the array length, the program fails",
+	"extract3":      "pop a byte-array A and two integers B and C. Extract a range of bytes from A starting at B up to but not including B+C, push the substring result. If B or B+C is larger than the array length, the program fails",
+	"extract16bits": "pop a byte-array A and integer B. Extract a range of bytes from A starting at B up to but not including B+2, push the substring result. If B or B+2 is larger than the array length, the program fails",
+	"extract32bits": "pop a byte-array A and integer B. Extract a range of bytes from A starting at B up to but not including B+4, push the substring result. If B or B+4 is larger than the array length, the program fails",
+	"extract64bits": "pop a byte-array A and integer B. Extract a range of bytes from A starting at B up to but not including B+8, push the substring result. If B or B+8 is larger than the array length, the program fails",
```

Comment:
For these I think we also want to convert the 2, 4, or 8 bytes to uint.  I can convinced otherwise, but it seems the intent is to make int extract easier.

Reply:
My bad, the spec doc is wrong -- will change so that it says to push an int